### PR TITLE
PEO-1150 Update CircleCI configuration

### PR DESCRIPTION
**Pull request by WoLkErSs**
[PEO-1150](https://garage.atlassian.net/browse/PEO-1150)

### Description

Update circleci config

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project
- [x] Performed a self-review of own code
- [ ] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,114 +1,89 @@
...
-      - store_artifacts:
-          path: ~/boilerplate-rails-api/coverage
-          destination: coverage
+      - attach_workspace:
+          at: ~/boilerplate-rails-api
+      - restore_cache:
+          keys:
+            - boilerplate-rails-api-{{ checksum "Gemfile.lock" }}
+            - boilerplate-rails-api-
+      - run: gem i bundler -v $(tail -1 Gemfile.lock | tr -d ' ')
```
Comment (chubchenko) [inline .circleci/config.yml:34]: Are you sure that we need this line?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,114 +1,89 @@
...
+          paths:
+            - vendor/bundle
+  overcommit:
+    executor: default
+    steps:
+      - attach_workspace:
+          at: ~/boilerplate-rails-api
+      - run: sudo apt-get install cmake
+      - run: |
+          echo 'export BUNDLER_VERSION=$(cat Gemfile.lock | tail -1 | tr -d " ")' >> $BASH_ENV
```
Comment (chubchenko) [inline .circleci/config.yml:53]: Same question

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,114 +1,89 @@
...
+          gem install bundler
+      - run: bundle install
+      - run: |
+          bundle exec overcommit -s
+          SKIP=AuthorEmail,AuthorName bundle exec overcommit -r
+  rspec:
+    executor: default
+    docker:
+      - image: circleci/ruby:2.6.5-node-browsers
+      - image: circleci/postgres:11-alpine
```
Comment (chubchenko) [inline .circleci/config.yml]: Don't forget to add `-ram`

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,114 +1,89 @@
...
+          source $BASH_ENV
+          gem install bundler
+      - run: bundle install
+      - run: |
+          bundle exec overcommit -s
+          SKIP=AuthorEmail,AuthorName bundle exec overcommit -r
+  rspec:
+    executor: default
+    docker:
+      - image: circleci/ruby:2.6.5-node-browsers
```
Comment (chubchenko) [inline .circleci/config.yml]: Why do we need `-browsers` here?

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,114 +1,89 @@
...
+      - image: circleci/postgres:11-alpine
+        environment:
+          - POSTGRES_USER: boilerplate
+    parallelism: 2
+    steps:
+      - attach_workspace:
+          at: ~/boilerplate-rails-api
+      - run: |
+          echo 'export BUNDLER_VERSION=$(cat Gemfile.lock | tail -1 | tr -d " ")' >> $BASH_ENV
```
Comment (chubchenko) [inline .circleci/config.yml:72]: Same question

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,114 +1,89 @@
...
-      - RAILS_ENV: test
-      - DB_USERNAME: root
-      - DB_PASSWORD: circle_test
-
+      - run: cp config/database.yml.ci config/database.yml
+      - persist_to_workspace:
+          root: ~/boilerplate-rails-api
+          paths:
+            - .
+  bundle:
```
Comment (chubchenko) [inline .circleci/config.yml:25]: Don't forget about gaps between jobs

**Review by chubchenko — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 4 above)
Comment (WoLkErSs) [reply .circleci/config.yml]: We will use it when add active_admin and feature specs

File: .circleci/config.yml [hunk 7]
```diff
@@ -1,114 +1,92 @@
...
-jobs:
-  linters:
-    <<: *defaults
+version: 2.1
 
+executors:
+  default:
+    description: The official CircleCI Ruby Docker image
+    docker:
+      - image: circleci/ruby:2.6.5-node-browsers
```
Comment (chubchenko) [inline .circleci/config.yml]: Why do we need `browsers`?

**Pull request merged**